sycl/doc/extensions/proposed/sycl_ext_oneapi_forward_progress.asciidoc

```diff
@@ -0,0 +1,503 @@
+- All threads of execution spawned from the same sub-group thread of execution
+  are considered part of the same sub-group scope.
+- Each individual work-item thread of execution is the sole member of its own
+  work-item scope.
+
+The forward progress guarantee of a thread of execution at scope `S`, relative
+to coordination scope `C`, is the weakest of the forward progress guarantees
+provided by threads of execution at all scopes between `S` and `C` inclusive.
+
+[NOTE]
+====
+Qualifying a thread of execution's forward progress guarantees with a coordination
+scope can be thought of as a shorthand for combining the forward progress
+guarantees provided by each thread of execution in the hierarchy.
+
+For example, consider a hypothetical implementation that creates a separate
+`std::thread` (providing concurrent forward progress guarantees) to execute
+each sub-group, but which permits an unbounded number of work-groups executed
+as tasks (providing parallel forward progress guarantees). With work-group
+coordination scope, each sub-group provides concurrent forward progress
+guarantees: this reflects the guarantees associated with each `std::thread`
+upon creation. With root-group coordination scope, each sub-group provides
+only parallel forward progress guarantees: each sub-group's progress guarantees
+are weakened to reflect that the creation of a sub-group depends upon a
+specific work-group (task) being scheduled and making progress.
+====
+
+[NOTE]
+====
+Choosing a coordination scope is similar to choosing a memory scope for atomic
+operations, in that it describes the potential set of threads of execution
+that a thread _may_ attempt to coordinate with, and must be chosen to reflect
+a kernel's dynamic communication pattern rather than static properties of its
+source code.
+
+For example, consider a kernel using locks/mutexes. Attempting to acquire a
+lock is a blocking operation, and so if two threads of execution providing
+weakly parallel forward progress guarantees attempt to acquire the lock
+simultaneously this may result in a deadlock. To correctly avoid a deadlock,
+the developer must correctly identify the coordination scope: if any work-item
+may try to acquire the lock, then work-items require stronger forward progress
+guarantees at root-group coordination scope; if each work-group has its own
+private data structure(s), and only work-items within the same work-group may
+try to acquire the same lock, then work-items require stronger forward progress
+guarantees only at work-group coordination scope.
+
+For performance reasons, a coordination scope will typically be the narrowest
+scope containing all threads that may coordinate with one another.
+====
+
+=== Forward progress guarantees
+
+The `forward_progress_guarantee` `enum` is used to represent the three classes
+of forward progress guarantee defined in the {cpp} core language specification.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental::sycl {
```

Comment:
Should this (and other) namespace definition be `sycl::ext::oneapi::experimental` instead? (`sycl::` in the beginning and not at the end)

Reply:
...Yes.  I have no idea how this happened!  Probably a bad vim search and replace.  Good catch!